### PR TITLE
fix(ct/core): update locale links

### LINKED ISSUE #5886: Urls are not redirecting to localized pages
Some routings does not include localized paths. When customer changes their language, e.g. from English to German (de), some links still have default routes, e.g. `/c/category` instead of `/de/c/category`

### Expected result
All urls should redirect to localized pages

### Actual result
Some links redirect to default routes.
Those links being:

* Go to Checkout on Cart Sidebar
* My profile - My Account icon in header
* Sidebar links in My Account
* … might be more

### Steps to reproduce
1. Go to homepage
1. Click language icon in the header
1. Change language, e.g. to German
1. Login in as a customer
1. Click My Account icon in header

### PR DESCRIPTION
**Pull request by Baroshem**
### Related Issues

closes #5886 

### Pull Request Checklist

- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- I tested manually my code and it works well with both:
- [ ] Default Theme
- [ ] Capybara Theme
- [ ] I have written test cases for my code

- [ ] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

File: packages/commercetools/theme/pages/Checkout/Billing.vue [hunk 1]
```diff
@@ -226,7 +226,7 @@
             {{ $t('Continue to payment') }}
           </SfButton>
           <nuxt-link
-            to="/checkout/shipping"
+            :to="localePath('/checkout/shipping')"
```
Comment (Mateuszpietrusinski) [inline packages/commercetools/theme/pages/Checkout/Billing.vue:229]: even better if there will be posibility to use object
```suggestion
            :to="localePath({name: 'shipping' })"
```

File: packages/commercetools/theme/pages/Checkout/Payment.vue [hunk 2]
```diff
@@ -199,7 +199,7 @@ export default {
 
     const processOrder = async () => {
       await make();
-      context.root.$router.push(`/checkout/thank-you?order=${order.value.id}`);
+      context.root.$router.push(context.root.localePath(`/checkout/thank-you?order=${order.value.id}`));
```
Comment (Mateuszpietrusinski) [inline packages/commercetools/theme/pages/Checkout/Payment.vue:202]:
```suggestion
      const path = {name: 'thank-you', query: {order: order.value.id}}
      context.root.$router.push(context.root.localePath(path));
```

**Review by Mateuszpietrusinski — APPROVED** [review]
Probably works ok, but think about replacing string paths via object for route.

File: packages/commercetools/theme/pages/Checkout/Payment.vue (hunk 2 above)
Comment (bloodf) [reply packages/commercetools/theme/pages/Checkout/Payment.vue:202]: I like the @Mateuszpietrusinski sugestion

**Review by bloodf — COMMENTED** [review]
I think if possible to use with object for paths, can be a cleaner solution for the future :)

File: packages/core/nuxt-theme-module/theme/pages/MyAccount.vue [hunk 3]
```diff
@@ -98,11 +98,14 @@ export default {
     const changeActivePage = async (title) => {
       if (title === 'Log out') {
         await logout();
-        $router.push('/');
+        $router.push(context.root.localePath({ name: 'home' }));
         return;
       }
 
-      $router.push(`/my-account/${(title || '').toLowerCase().replace(' ', '-')}`);
+      const transformedPath = `/my-account/${(title || '').toLowerCase().replace(' ', '-')}`;
```
Comment (Fifciu) [inline packages/core/nuxt-theme-module/theme/pages/MyAccount.vue:105]:
```suggestion
      const slugifiedTitle = (title || '').toLowerCase().replace(' ', '-');
      const transformedPath = `/my-account/${slugifiedTitle}`;
```

**Review by Fifciu — CHANGES_REQUESTED** [review]

**Review by Fifciu — APPROVED** [review]

**Pull request merged**